ReactAndroid/src/main/java/com/facebook/react/modules/network/OkHttpClientProvider.java

```diff
@@ -39,11 +47,43 @@ public static void replaceOkHttpClient(OkHttpClient client) {
 
   private static OkHttpClient createClient() {
     // No timeouts by default
-    return new OkHttpClient.Builder()
+    OkHttpClient.Builder client = new OkHttpClient.Builder()
       .connectTimeout(0, TimeUnit.MILLISECONDS)
       .readTimeout(0, TimeUnit.MILLISECONDS)
       .writeTimeout(0, TimeUnit.MILLISECONDS)
-      .cookieJar(new ReactCookieJarContainer())
-      .build();
+      .cookieJar(new ReactCookieJarContainer());
+
+    return enableTls12OnPreLollipop(client).build();
   }
+
+  /*
+    On Android 4.1-4.4 (API level 16 to 19) TLS 1.1 and 1.2 are
+    available but not enabled by default. The following method
+    enables it.
+
+    See also https://github.com/facebook/react-native/issues/7192
+   */
+  public static OkHttpClient.Builder enableTls12OnPreLollipop(OkHttpClient.Builder client) {
+    if (Build.VERSION.SDK_INT >= 16 && Build.VERSION.SDK_INT < 20) {
```

Comment:
Use `Build.VERSION_CODES` for api levels

`Build.VERSION_CODES.JELLY_BEAN` == 16
`Build.VERSION_CODES.KITKAT_WATCH ` == 20 - maybe use `KITKAT` (v19) with a `<=` for readability